### PR TITLE
DOCS-11661: Add a warning to discourage people from setting failIndexKeyTooLong

### PR DESCRIPTION
**Pull request by steveren**
Staged: https://docs-mongodbcom-staging.corp.mongodb.com/steverenaker/DOCS-11661/reference/parameters.html#param.failIndexKeyTooLong

### CODE REVIEW

File: source/reference/parameters.txt [hunk 1]
```diff
@@ -558,6 +558,13 @@ General Parameters
    upgrade and then gradually resolve these indexing issues, you can
    use :parameter:`failIndexKeyTooLong` to disable this behavior.
 
+   .. important::
+
+      Setting :parameter:`failIndexKeyTooLong` to ``false`` is
+      intended to be a temporary workaround. Leaving
```
Comment (rob-mongodb) [inline source/reference/parameters.txt]: Nit: Should we explicitly state what problem is solved by the workaround?

Comment (steveren) [reply source/reference/parameters.txt]: I think it's adequately explained in the paragraphs above, but if you think the explanation should go in the note as well I can add it there.

File: source/reference/parameters.txt [hunk 2]
```diff
@@ -558,6 +558,14 @@ General Parameters
    upgrade and then gradually resolve these indexing issues, you can
    use :parameter:`failIndexKeyTooLong` to disable this behavior.
 
+   .. important::
+
+      Setting :parameter:`failIndexKeyTooLong` to ``false`` is
+      intended to be a temporary workaround, not a permanent
+      solution to the problem of oversized index keys. Leaving
+      indexes in place which contain oversized keys is not
+      recommended, because they may return incomplete results.
```
Comment (rob-mongodb) [inline source/reference/parameters.txt]: Superfluous comma.

File: source/reference/parameters.txt [hunk 3]
```diff
@@ -558,6 +558,14 @@ General Parameters
    upgrade and then gradually resolve these indexing issues, you can
    use :parameter:`failIndexKeyTooLong` to disable this behavior.
 
+   .. important::
+
+      Setting :parameter:`failIndexKeyTooLong` to ``false`` is
+      intended to be a temporary workaround, not a permanent
```
Comment (rob-mongodb) [inline source/reference/parameters.txt]: Consider deleting "is intended to be" from this sentence.

**Pull request merged**